### PR TITLE
Improve syntax highlighting

### PR DESCRIPTION
**Pull request by monkoose**
Various improvements to syntax highlighting.

More info in the commit messages or in the comments.

### CODE REVIEW

File: syntax/fennel.vim [hunk 1]
```diff
@@ -9,7 +9,7 @@ endif
 let s:cpo_sav = &cpo
 set cpo&vim
 
-if has("folding") && exists("g:fennel_fold") && g:fennel_fold > 0
+if get(g:, 'fennel_fold') > 0
```
Comment (monkoose) [inline syntax/fennel.vim:12]: We don't need to support vim right in a neovim only plugin, right?

File: syntax/fennel.vim [hunk 2]
```diff
@@ -17,10 +17,14 @@ endif
 sy match FennelComment ";.*$" contains=FennelCommentTodo,@Spell display
 sy keyword FennelCommentTodo contained FIXME XXX TODO FIXME: XXX: TODO:
 
-
-syntax match FennelStringEscape '\v\\%([abfnrtv'"\\]|x[[0-9a-fA-F]]\{2}|25[0-5]|2[0-4][0-9]|[0-1][0-9][0-9])' contained
-syntax region FennelString matchgroup=FennelStringDelimiter start=/"/ skip=/\\\\\|\\"/ end=/"/ contains=FennelStringEscape,@Spell
-syntax region FennelString matchgroup=FennelStringDelimiter start=/'/ skip=/\\\\\|\\'/ end=/'/ contains=FennelStringEscape,@Spell
```
Comment (monkoose) [inline syntax/fennel.vim:23]: It is a compilation error to use single quoted strings. Or am I miss something?

**Comment by monkoose** [comment]
Not sure about naming conventions for highlight groups. But don't think it is too much important.

Maybe some highlights are questionable for those who accustomed to simpler one. Open for a discussion.

**Comment by monkoose** [comment]
Most work that I have wanted to do is done.
There is some todo i left in the file, but it is for another PR some time lately.

File: syntax/fennel.vim (hunk 1 above)
Comment (Olical) [reply syntax/fennel.vim:19]: Agreed

File: syntax/fennel.vim (hunk 2 above)
Comment (Olical) [reply syntax/fennel.vim:23]: Yeah I think you're right, single quote is used to denote a symbol like in Clojure. Maybe this used to be a thing in old versions of the language.

**Comment by Olical** [comment]
Looks good to me although as I've said before, I'm no expert with this sort of thing so I'm leaning on your experience 😃 I'll add something to the help text mentioning the `g:` values for you. I'll merge to `develop` and we can test drive it for a while.

**Comment by Olical** [comment]
Thank you for working on this!

**Pull request merged**